webui/src/pages/repositories/repository/branches.jsx

```diff
@@ -56,63 +60,73 @@ const BranchWidget = ({ repo, branch, onDelete }) => {
     }
 
     return (
-        <ListGroup.Item>
+        <ListGroup.Item style={selected ? { backgroundColor: 'rgba(0, 123, 255, 0.1)' } : {}}>
             <Row className="d-flex align-items-center justify-content-between">
+                <Col md="auto" className="d-flex align-items-center">
+                    <Checkbox
+                        key={`${branch.id}-${selected}`}
+                        name={branch.id}
+                        disabled={isDefault}
+                        defaultChecked={selected}
+                        onAdd={() => onSelect && onSelect(branch.id)}
+                        onRemove={() => onDeselect && onDeselect(branch.id)}
+                    />
+                </Col>
```

Comment:
It looks like we will need this functionality in every other list, too.  Can we extract it to a separate component?